lib/services/front.ts

```diff
@@ -1,357 +1,127 @@
-	 * @returns         Promise that resolves to the user id.
-	 */
-	private fetchUserId = (username: string): Promise<string|undefined> => {
-		// Request a list of all teammates
-		const getTeammates = {
-			headers: {
-				authorization: `Bearer ${this.data.token}`
-			},
-			json: true,
-			method: 'GET',
-			uri: 'https://api2.frontapp.com/teammates',
-		};
-		return request(getTeammates).then((teammates: {_results: Array<{username: string, id: string}>}) => {
-			// Resolve to the ID of the first matching teammate
-			const teammate = _.find(teammates._results, (eachTeammate) => {
-				return eachTeammate.username === username;
-			});
-			if (teammate) {
-				return teammate.id;
-			}
-		});
-	}
-
-	/**
-	 * Attempt to find a recent conversation ID from it's subject line.
-	 * Done by subject because the conversation_reference provided is sometimes junk.
-	 * @param subject       Target subject line to search for.
-	 * @param attemptsLeft  Since conversations take time to propagate this method may recurse.
-	 * @returns             Promise that resolves to the ID of the conversation.
-	 */
-	private findConversation = (subject: string, attemptsLeft: number = 10): Promise<string> => {
-		// Find all the recent conversations
-		return this.session.conversation.list().then((response) => {
-			// Filter these down to matching conversations
-			const conversationsMatched = _.filter(response._results, (conversation) => {
-				return conversation.subject === subject;
-			});
-			// Return the most recent, if any
-			if (conversationsMatched.length > 0) {
-				return conversationsMatched[0].id;
-			}
-			// Recurse up to the specified number of times
-			if (attemptsLeft > 1) {
-				return this.findConversation(subject, attemptsLeft - 1);
-			}
-			throw new Error('Could not find relevant conversation.');
-		});
-	}
-
-	/**
-	 * The name of this service, as required by the framework.
-	 * @returns  'flowdock' string.
-	 */
-	get serviceName(): string {
-		return FrontService._serviceName;
-	}
-
-	/**
-	 * Retrieve the SDK API handle for Front.
-	 * @return  The Front SDK API handle.
-	 */
-	get apiHandle(): FrontHandle {
-		return {
-			front: this.session
-		};
-	}
-}
-
-/**
- * Build this class, typed and activated as a listener.
- * @returns  Service Listener object, awakened and ready to go.
- */
-export function createServiceListener(data: FrontConstructor): ServiceListener {
-	return new FrontService(data, true);
-}
-
-/**
- * Build this class, typed as an emitter.
- * @returns  Service Emitter object, ready for your events.
- */
-export function createServiceEmitter(data: FrontConstructor): ServiceEmitter {
-	return new FrontService(data, false);
-}
-
-/**
- * Build this class, typed as a message service.
- * @returns  Message Service object, ready to convert events.
- */
-export function createMessageService(data: FrontConstructor): Messenger {
-	return new FrontService(data, false);
-}
+// /*
```

Comment:
The Front service has currently been put temporarily out of scope, as indicated by the massive commenting, and will be coming back into the fold in a couple of days.